### ISSUE TITLE
Inconsistent failover when there is no failover timeout set

### ISSUE THREAD

**Issue by aniketschneider** [flyerhzm/redis-sentinel#39]
This bug occurs if there is no `:failover_reconnect_timeout` configured and a failover occurs, but the client is still able to connect to the old master node (which is now a slave).  The client will stay connected to the old master node and writes will fail with READONLY errors.

This situation can be easily replicated by setting up sentinels as in the example, but changing the example script to write instead of just reading.  Run the example script, then issue `sentinel failover example-test` to one of the sentinel nodes.

My take on the bug is as follows:
- when the client can't connect at all to the server, a ConnectionError is raised and a reconnect is initiated (via code from the redis-rb gem).
- when the client attempts a write and can connect to the server, but the server is a READONLY slave, a CommandError is raised instead.
- if there is no reconnect timeout configured, or the reconnect timeout is 0, [this check](https://github.com/flyerhzm/redis-sentinel/blob/master/lib/redis-sentinel/client.rb#L148) raises the error back to the base redis client without attempting to reconnect
- because it is a CommandError, the code in the redis-rb gem does not try to reconnect, so the client remains connected to the slave.

Basically if you don't have a failover_reconnect_timeout configured, or if you have it configured to 0, the client gives up attempting to reconnect without trying to reconnect even once.
